### PR TITLE
Refactoring + improvements in retrieving peppy project + fixed few bugs

### PR DESCRIPTION
**Pull request by Khoroshevskyi**
PR after refactoring of the code.
Things that have been done:
- Refactoring
- Added docstrings to the methods
- Added user documentation
- Fixed bug, where yaml files weren't creating
- New written method for saving peps for raw data
- Fixed minor bugs in creating peppy Projects

TODO:

- [x] peppy new version should be added after it's release
- [x] Change name get_project to get_projects
- [x] Add gse number to every output of get_projects.
- [x] Add Finder to geofetch

**Reviewers**: Could you please check all the documentation. I have added few new pages and updated old one.

- [ ] Checking documentation
- ~[ ] Add codecov (I (Alex) don't know where is an error and how to solve it)~ - Won't be added as it's quite hard to test methods that interact with online databases

### CODE REVIEW

**Comment by Khoroshevskyi** [comment]
- [ ] Change name get_project to get_projects

- [ ] Add gse number to every output of get_projects.

**Comment by Khoroshevskyi** [comment]
- [ ] Add Finder to geofetch

**Comment by Khoroshevskyi** [comment]
@nleroy917, @nsheff I have corrected preveous issues that were discussed here and added documentation. Geofetch is ready for the review.

**Comment by nsheff** [comment]
Are you intending for this to be v0.11.0?

**Comment by Khoroshevskyi** [comment]
> Are you intending for this to be v0.11.0?

Yes, because new features were added: Python API and Finder

File: geofetch/geofetch.py [hunk 1]
```diff
@@ -31,14 +33,72 @@
...
+
+    :param split_experiments: Split SRR runs into individual samples. By default, SRX experiments with multiple SRR
+            Runs will have a single entry in the annotation table, with each run as a separate row in the
+            subannotation table. This setting instead treats each run as a separate sample [Works with raw data]
+    :param bam_folder: Optional: Specify folder of bam files. Geofetch will not download sra files when
+            corresponding bam files already exist. [Default: $SRABAM:] [Works with raw data]
+    :param fq_folder: Optional: Specify folder of fastq files. Geofetch will not download sra files when corresponding
+            fastq files already exist. [Default: $SRAFQ:] [Works with raw data]
+    :param use_key_subset: Use just the keys defined in this module when writing out metadata. [Works with raw data]
+    :param sra_folder: [Doesn't work ]Optional: Specify a location to store sra files
```
Comment (nsheff) [inline geofetch/geofetch.py:93]: Can you clarify this "Doesn't work" issue? Is there something wrong?

File: geofetch/geofetch.py [hunk 2]
```diff
@@ -331,443 +396,477 @@ def fetch_all(self, input: str, name: str = None):
...
+                            "Bam conversion failed with sam-dump. Trying fastq-dump..."
+                        )
+                        self._sra_bam_conversion2(bam_file, run_name, self.picard_path)
+
+                except FileNotFoundError as err:
+                    self._LOGGER.info(
+                        f"SRA file doesn't exist, please download it first: {err}"
+                    )
+
+    def fetch_processed_one(
```
Comment (nsheff) [inline geofetch/geofetch.py:633]: I think the usual way to say this would be `fetchone_processed`

File: geofetch/geofetch.py [hunk 3]
```diff
@@ -331,443 +396,477 @@ def fetch_all(self, input: str, name: str = None):
...
+        meta_processed_series = self._standardize_colnames(meta_processed_series)
+
+        return meta_processed_samples, meta_processed_series
+
+    def _generate_processed_meta(
+        self, name: str, meta_processed_samples: list, meta_processed_series: list
+    ) -> dict:
+        """
+        Generate and save PEPs for processed accessions. GEO has data in GSE and GSM,
+            conditions are used to decide which PEPs have to be saved.
```
Comment (nsheff) [inline geofetch/geofetch.py:673]: maybe "will be saved"?

**Review by nsheff — APPROVED** [review]

File: geofetch/geofetch.py (hunk 1 above)
Comment (Khoroshevskyi) [reply geofetch/geofetch.py:93]: It works. But this argument is not added to parser. Already fixed it! And it works

**Pull request merged**